src/Controller/EnMarche/UserController.php

```diff
@@ -26,13 +28,34 @@ class UserController extends Controller
 
     /**
      * @Route("", name="app_user_profile")
-     * @Method("GET|POST")
+     * @Method("GET")
      */
     public function profileOverviewAction(): Response
     {
         return $this->render('user/my_account.html.twig');
     }
 
+    /**
+     * @Route("/mes-dons", name="app_user_profile_donation")
+     * @Method("GET")
+     */
+    public function profileDonationAction(): Response
+    {
+        $donationRepos = $this->getDoctrine()->getRepository(Donation::class);
+        /** @var Adherent $user */
+        $user = $this->getUser();
+
+        $lastDonation = null;
+        if ($donations = $donationRepos->findBy(['emailAddress' => $user->getEmailAddress()])) {
+            $lastDonation = $donations[0];
+        }
+
+        return $this->render('user/my_donation.html.twig', [
```

Comment:
Don't use this logic in the controller.
The template should do that.

e.g.:
```
public function profileDonationAction(): Response
{
    $donationRepository = $this->getDoctrine()->getRepository(Donation::class);
    $user = $this->getUser();

    return $this->render('user/my_donation.html.twig', [
        'donations' => $donationRepository->findBy(['emailAddress' => $user->getEmailAddress()]),
    ]);
}
```

Then in the template you can simply retrieve the first elment with : `donations|first`
see https://twig.symfony.com/doc/2.x/filters/first.html